### PR TITLE
AV-86846  handle payload in post

### PR DESCRIPTION
**Pull request by manojkumarjain**
AV-86846  handle payload in post

### CODE REVIEW

File: go/session/avisession.go [hunk 1]
```diff
@@ -950,6 +950,8 @@ func (avisess *AviSession) GetRaw(uri string) ([]byte, error) {
 
 // PostRaw performs a POST API call and returns raw data
 func (avisess *AviSession) PostRaw(uri string, payload interface{}) ([]byte, error) {
+        glog.Infof("MANOJ PostRaw: %v", payload)
```
Comment (grastogi23) [inline go/session/avisession.go]: please remove "Manoj" from the debug. Also, printing payload may be very expensive. Let us also add unit test for this. I am surprised this was not caught in the previous commit.

**Review by grastogi23 — CHANGES_REQUESTED** [review]

File: go/session/avisession.go (hunk 1 above)
Comment (manojkumarjain) [reply go/session/avisession.go]: Will do after testing on ctrl

Comment (grastogi23) [reply go/session/avisession.go]: let us may be create a jira for sandeep or sdk to add support for PutRaw as well. I checked that is not supported.

File: go/session/avisession.go [hunk 2]
```diff
@@ -950,6 +950,8 @@ func (avisess *AviSession) GetRaw(uri string) ([]byte, error) {
...
+        resp, rerror := avisess.restRequest("POST", uri, payload, "", nil)
 	resp, rerror := avisess.restRequest("POST", uri, nil, "", nil)
```
Comment (chauhanshubham) [inline go/session/avisession.go:955]: I think replacing nil with payload here is the only thing required. Newline at L954 needs to be replaced at L955

File: go/session/avisession.go [hunk 3]
```diff
@@ -950,7 +950,8 @@ func (avisess *AviSession) GetRaw(uri string) ([]byte, error) {
 
 // PostRaw performs a POST API call and returns raw data
 func (avisess *AviSession) PostRaw(uri string, payload interface{}) ([]byte, error) {
-	resp, rerror := avisess.restRequest("POST", uri, nil, "", nil)
+        glog.Infof("PostRaw: %v", payload)
```
Comment (chauhanshubham) [inline go/session/avisession.go]: not sure if we should log this, considering we are not logging the Post() payload as well. Maybe change it to Debug or remove it for consistency across other functions

File: go/session/avisession.go (hunk 1 above)
Comment (manojkumarjain) [reply go/session/avisession.go]: Sure Gaurav

**Comment by manojkumarjain** [comment]
@grastogi23  please approve and merge.
Unit testing completed

**Review by chauhanshubham — APPROVED** [review]

**Review by grastogi23 — APPROVED** [review]

**Pull request merged**